numpy/random/tests/test_extending.py

```diff
@@ -22,30 +24,35 @@
 
 try:
     import cython
+    from Cython.Compiler.Version import version as cython_version
 except ImportError:
     cython = None
+else:
+    from distutils.version import LooseVersion
+    # Cython 0.29.14 is required for Python 3.8 and there are
+    # other fixes in the 0.29 series that are needed even for earlier
+    # Python versions.
+    # Note: keep in sync with the one in pyproject.toml
+    required_version = LooseVersion('0.29.14')
+    if LooseVersion(cython_version) < required_version:
+        # too old or wrong cython, skip the test
+        cython = None
 
 @pytest.mark.skipif(cython is None, reason="requires cython")
-def test_cython():
-    curdir = os.getcwd()
-    argv = sys.argv
-    examples = (os.path.dirname(__file__), '..', '_examples')
-    try:
-        os.chdir(os.path.join(*examples))
-        sys.argv = argv[:1] + ['build']
-        with warnings.catch_warnings(record=True) as w:
-            # setuptools issue gh-1885
-            warnings.filterwarnings('always', '', DeprecationWarning)
-            from numpy.random._examples.cython import setup
-    finally:
-        sys.argv = argv
-        os.chdir(curdir)
+@pytest.mark.slow
+@pytest.mark.skipif(sys.platform == 'win32', reason="cmd too long on CI")
```

Comment:
Do you have a link to a CI failure for this? It's very weird that a command line with only 3 tokens in it would fail here.

Reply:
This one?

https://dev.azure.com/numpy/numpy/_build/results?buildId=7086&view=ms.vss-test-web.build-test-results-tab&runId=1084568&resultId=109270&paneView=debug

Seems strange that we're not capturing any subprocess output.

Reply:
Error is [here](https://dev.azure.com/numpy/numpy/_build/results?buildId=7086&view=logs&j=41da7a62-01d8-5a8a-1897-893bcc993eb5&t=6e2ca7fa-7d8b-55ee-d2ce-056d635fc0b4&l=263):

```


creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp

creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator

creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0

creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0

creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples

creating build\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples\cython

C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DNPY_NO_DEPRECATED_API=0 -IC:\hostedtoolcache\windows\Python\3.7.5\x64\lib\site-packages\numpy\core\include -IC:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples\cython\..\.. -IC:\hostedtoolcache\windows\Python\3.7.5\x64\include -IC:\hostedtoolcache\windows\Python\3.7.5\x64\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\include" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\cppwinrt" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.17763.0\winrt" /TcC:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples\cython\extending.c /Fobuild\temp.win-amd64-3.7\Release\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples\cython\extending.obj

extending.c

C:\Users\VssAdministrator\AppData\Local\Temp\pytest-of-VssAdministrator\pytest-0\test_cython0\_examples\cython\extending.c : fatal error C1083: Cannot open compiler generated file: '': Invalid argument
```

Reply:
I suspect removing the `abspath` on this line will fix it:

https://github.com/numpy/numpy/blob/c18b3f59b5f5752ab038224fd3bdf2dadba55ecf/numpy/random/_examples/cython/setup.py#L14